src/Domain/Direction.hpp

```diff
@@ -0,0 +1,153 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines class template Direction.
+
+#pragma once
+#pragma GCC diagnostic ignored "-Wpedantic"
+
+#include <functional>
+#include <iosfwd>
+#include <pup.h>
+
+#include "Domain/Side.hpp"
+#include "Parallel/PupStlCpp11.hpp"
+
+/// \ingroup ComputationalDomain
+/// A particular Side along a particular coordinate Axis.
+template <size_t VolumeDim>
+class Direction {
+ public:
+  /// The logical-coordinate names of each dimension
+  enum class Axis;
+
+  /// Construct by specifying an Axis and a Side.
+  Direction(Axis axis, Side side) : axis_(axis), side_(side) {}
+
+  /// Default constructor for Charm++ serialization.
+  Direction() = default;
+
+  /// The dimension of the Direction
+  size_t dimension() const { return static_cast<size_t>(axis_); }
+
+  /// The Axis of the Direction
+  Axis axis() const { return axis_; }
+
+  /// The side of the Direction
+  Side side() const { return side_; }
+
+  /// The sign for the normal to the Side.
+  double sign() const { return (Side::Lower == side_ ? -1.0 : 1.0); }
+
+  /// The opposite Direction.
+  Direction<VolumeDim> opposite() const;
+
+  /// Create a Direction from a dimension and a Side.
+  static Direction<VolumeDim> create_direction(size_t dimension, Side side);
+
+  // {@
+  /// Helper functions for creating specific Directions.
+  /// These are labeled by the logical-coordinate names (Xi,Eta,Zeta).
+  static Direction<VolumeDim> lower_xi();
```

Comment:
Since the Axis can be accessed publicly, these function should be tested explicitly